bin/bout-v5-input-file-upgrader.py

```diff
@@ -34,6 +37,12 @@ def fix_replacements(replacements, options_file):
                 "Could not apply transformation: '{old}' -> '{new}' to file '{0}', due to error:"
                 "\n\t{1}".format(options_file.filename, e.args[0], **replacement)
             ) from e
+        else:
+            if "values" in replacement:
+                old_value = options_file[replacement["new"]]
+                options_file[replacement["new"]] = replacement["values"].get(
+                    old_value, old_value
```

Comment:
@ZedThree should this be
```suggestion
                    old_value.lower(), old_value
```
?

Reply:
Good question! `options` is case insensitive for keys, I don't think it is for values, although BOUT++ _is_. I think `.lower()` is correct.

I'm a little worried about failure modes here. We might not want to use `dict.get` here, and instead `try... except` and print a nice error message instead?

Reply:
I'd intended to allow replacing just some of the possible values of an option, so if the old value isn't in replacments["values"], then we just keep the old value, so no need for an error message. This 'feature' isn't actually used now because in fixing the `bool` issue I've required both `True` and `False` to be present for bool-type options; if it turns out it's better to actually have an error, then we can change it.